docs/source/installing.rst

```diff
@@ -20,15 +19,68 @@ At the minimum, you will need:
 
 .. important::
 
-    If your computer does not have Python, or your Python version does not meet the requirement, you
-    must `update Python <https://www.python.org/downloads/>`_.
-
     To check the Python version::
 
         python --version
 
-    You will also need to install ``git`` for the platform on which you are about to run TimeChart. Follow these
-    `Git Installation Instructions <https://git-scm.com/book/en/v2/Getting-Started-Installing-Git>`_.
+    Note that if you have both Python 2 and Python 3 on your computer, and you want to check for the Python 3 version::
+
+        python3 --version
+
+    If your computer does not have Python, or your Python version does not meet the requirement, you
+    must `install or update Python <https://www.python.org/downloads/>`_.
+
+Operating Systems
+=================
+* Linux: RHEL 6.10+, CentOS 7.5+
+* macOS: High Sierra (10.13)+
+* Windows: Windows 10+, Windows 2016+
+
+
+.. _obtain the TimeChart source code:
+
+************************************
+Obtaining the TimeChart Source Code
+************************************
+
+Unless you are going to install TimeChart with conda (see :ref:`using conda`), the rest of the installation options
+require you to either download or clone the TimeChart source code from GitHub.
+
+
+Downloading the Stable Release Source Code
+===========================================
+
+You can download the source code of one of TimeChart's stable releases from the `GitHub TimeChart Release page
+<https://github.com/slaclab/timechart/releases>`_. Make sure you unzip or untar the downloaded file into a separate
+directory.
+
+In all the examples below, it is assumed that that directory's name is ``timechart``.
+
+
+Cloning the Source Code
+========================
+
+.. important::
+    If you opt to clone the TimeChart source code from GitHub, you will also need to install ``git`` for the
+    platform on which you are about to run TimeChart. Follow these
+    `Git Installation Instructions <https://git-scm.com/book/en/v2/Getting-Started-Installing-Git>`_ to install ``git``.
+
+#. Clone the TimeChart repo::
+
+    git clone https://github.com/slaclab/timechart.git
+
+#. If you want to build TimeChart with the latest code, you can stop here. Otherwise, to build TimeChart with source
+   code from a stable release, you will need to check out a release tag into a new branch, and then check out that
+   branch. All of these can be accomplished with one single command::
+
+    git checkout <tag> -b <new_branch_name>
+
+For example, if you want to build TimeChart with the stable release
+`v1.2.1 <https://github.com/slaclab/timechart/releases/tag/v1.2.1>`_, use this command::
+
+    git checkout v1.2.1 -b my_release_branch
```

Comment:
I don't want to bombard the user with too many options. Expert users would know what to do. But to simplify the steps, I just provide this option here -- checkout a tag into a separate branch. That's a clean option. Do you agree?

Reply:
No, this in fact add another option to a command that is not required. If the goal is to avoid confusion go simple and check the tag only, no need for new branch

Reply:
So, just tell them to do ``git checkout v1.2.1`` then?

Reply:
Yup 👍

Reply:
Here's another scenario: what if this user wants to checkout another tag? If they had a branch for each tag, they'd know what they're running, and can report issues for a specific release correctly.

I'm not seeing a confusing with a branch. Let's just keep this instruction.

Reply:
What you described is confusing already. Just check out the tag, no new branch. There is no need. If the person needs another tag they run a new checkout. They only need a branch if they intend to do changes and keep track but that is the case for developers and they know how to deal with it

Reply:
What you described is confusing already. Just check out the tag, no new branch. There is no need. If the person needs another tag they run a new checkout. They only need a branch if they intend to do changes and keep track but that is the case for developers and they know how to deal with it

Reply:
What you described is confusing already. Just check out the tag, no new branch. There is no need. If the person needs another tag they run a new checkout. They only need a branch if they intend to do changes and keep track but that is the case for developers and they know how to deal with it